### ISSUE TITLE
Understanding emotions - from Keras to pyTorch

### ISSUE THREAD

**Issue by MarioGzSl** [khuyentran1401/machine-learning-articles#27]
## TL;DR
How to implement DeepMoji from scratch in pyTorch.
 <img src="https://raw.githubusercontent.com/MarioGzSl/Images/master/deepemoji.png" alt="Paris" class="center"> 

### Article Link
[Understanding emotions - from Keras to pyTorch](https://medium.com/huggingface/understanding-emotions-from-keras-to-pytorch-3ccb61d5a983)
### Author 
Huggingface 🤗
## Key Takeaways
* Custom LSTM module
* Attention layer
* PackedSequence <a href="https://pytorch.org/docs/master/nn.html#pack-padded-sequence">[doc]</a> 
* DataSet class <a href="https://pytorch.org/docs/stable/data.html">[doc]</a> 
* Initialization of the weights

## Useful Code Snippets
```python
import torch

```

**Comment by khuyentran1401** [comment]
Nice article. What do you think about using this for?

**Comment by MarioGzSl** [comment]
This article is very useful in two ways.
- As a pipeline to develop NLP models in pyTorch from scratch
- For feature extraction related to emotions to feed other models.

**Comment by khuyentran1401** [comment]
Nice. I will go back to this article in the future  as I need to perform feature extraction related to emotion or want to build NLP models in PyTorch from scratch
